lib/AST/ASTPrinter.cpp

```diff
@@ -5050,6 +5098,13 @@ void PrintAST::visitErasureExpr(ErasureExpr *expr) {
 }
 
 void PrintAST::visitKeyPathExpr(KeyPathExpr *expr) {
+  auto components = expr->getComponents();
+  for (int i = 0; i <= (int) components.size() - 1; i++) {
+    printType(components[i].getComponentType());
```

Comment:
The component type isn't how the component is spelled in source, you'll need to switch over the different component kinds and implement the printing logic for each of the kinds.